### PR TITLE
T32507 basic navigation

### PR DESCRIPTION
**Pull request by manuq**
https://phabricator.endlessm.com/T32507

### CODE REVIEW

**Comment by manuq** [comment]
@danigm where are you getting those lint errors? Do you have the project setup with pipenv like it's explained in the README? From my last commit I don't see any errors:
```
$ pre-commit run --all-files
Trim Trailing Whitespace.....................................................Passed
Flake8.......................................................................Passed
Check Yaml...................................................................Passed
Check for added large files..................................................Passed
Debug Statements (Python)....................................................Passed
Fix End of Files.............................................................Passed
Reorder python imports.......................................................Passed
Linting of JS, Vue, SCSS and CSS files on the frontend.......................Passed
Linting of JS and Vue files on the packages..................................Passed
black........................................................................Passed
```

```
$ yarn lint-packages
yarn run v1.22.10
$ lerna run lint
lerna notice cli v4.0.0
lerna info versioning independent
lerna info Executing command in 4 packages: "yarn run lint"
lerna info run Ran npm script 'lint' in 'kolibri-api' in 1.0s:
$ eslint index.js
lerna info run Ran npm script 'lint' in 'eos-components' in 2.0s:
$ vue-cli-service lint
 DONE  No lint errors found!
lerna info run Ran npm script 'lint' in 'loading-screen' in 1.4s:
$ vue-cli-service lint
 DONE  No lint errors found!
lerna info run Ran npm script 'lint' in 'template-ui' in 2.0s:
$ vue-cli-service lint
 DONE  No lint errors found!
lerna success run Ran npm script 'lint' in 4 packages in 4.0s:
lerna success - eos-components
lerna success - kolibri-api
lerna success - loading-screen
lerna success - template-ui
Done in 4.62s.
```

File: packages/template-ui/src/views/ListSection.vue [hunk 1]
```diff
@@ -43,16 +43,57 @@ import { mapState, mapGetters } from 'vuex';
...
+      this.loadingSubsectionNodes = true;
+      this.subsectionNodes = {};
+      return Promise.all(this.sectionNodes.map((subsection) => {
+        return window.kolibri.getContentByFilter({ parent: subsection.id })
+          .then((page) => {
+            this.subsectionNodes[subsection.id] = page.results;
+          });
+      })).then(() => {
+        this.loadingSubsectionNodes = false;
+      });
```
Comment (danigm) [inline packages/template-ui/src/views/ListSection.vue:95]: Maybe the second `then` is not needed here, we can add the `this.loadingSubsectionNodes = false` inside the first one.

File: packages/template-ui/src/store/index.js [hunk 2]
```diff
@@ -71,7 +71,8 @@ const initialState = {
   hasDarkHeader: false,
   hasSectionsSearch: true,
   hasCarousel: true,
-  hasFilters: true,
+  // FIXME reenable the filters and migrate them to the API:
+  hasFilters: false,
```
Comment (danigm) [inline packages/template-ui/src/store/index.js:77]: Maybe we can keep some kind of filtering in the future, but just at leaf level, not with content inside folders.

**Review by danigm — APPROVED** [review]
Overall looks pretty good to me. I've been testing and it's working correctly. I've just left a few comments and added a new commit to fix some lint errors.

**Comment by danigm** [comment]
> @danigm where are you getting those lint errors? Do you have the project setup with pipenv like it's explained in the README? From my last commit I don't see any errors:

This is what I'm getting when I run the `yarn lint` inside the `template-ui` package:

```
HEAD está ahora en 8240ea1 template: Remove App routing
(kolibri-explore-plugin)[danigm@rincewind template-ui]!(HEAD $ yarn run lint
yarn run v1.22.10
$ vue-cli-service lint
error: Method 'fetchCarouselNodes' expected no return value (consistent-return) at src/views/Home.vue:123:9:
  121 |       this.loadingCarouselNodes = true;
  122 |       if (this.carouselNodeIds.length) {
> 123 |         return window.kolibri.getContentByFilter({ ids: this.carouselNodeIds })
      |         ^
  124 |           .then((page) => {
  125 |             this.carouselNodes = page.results;
  126 |             this.loadingCarouselNodes = false;


error: Method 'sectionNodes' expected no return value (consistent-return) at src/views/ListSection.vue:81:7:
  79 |         return;
  80 |       }
> 81 |       return this.fetchSubsectionNodes();
     |       ^
  82 |     },
  83 |   },
  84 |   methods: {


2 errors found.
error Command failed with exit code 1.
```

I don't know why this is happening to me and it's not happening in the CI or in your case. I think that I've my environment correctly set, but maybe I've something different, so feel free to remove the last commit.

File: packages/template-ui/src/views/ListSection.vue (hunk 1 above)
Comment (danigm) [reply packages/template-ui/src/views/ListSection.vue:95]: I've just noticed that this is a `Promise.all`, so forget about this comment.

**Comment by manuq** [comment]
> I don't know why this is happening to me and it's not happening in the CI or in your case. I think that I've my environment correctly set, but maybe I've something different, so feel free to remove the last commit.

That's very interesting. So it seems you have a correct environment. It's crazy that we are having different results:
```
$ git checkout 8240ea1
Nota: actualizando el árbol de trabajo '8240ea1'.

Te encuentras en estado 'detached HEAD'. Puedes revisar por aquí, hacer
cambios experimentales y confirmarlos, y puedes descartar cualquier
commit que hayas hecho en este estado sin impactar a tu rama realizando
otro checkout.

Si quieres crear una nueva rama para mantener los commits que has creado,
puedes hacerlo (ahora o después) usando -b con el comando checkout. Ejemplo:

  git checkout -b <nombre-de-nueva-rama>

HEAD está ahora en 8240ea1 template: Remove App routing
(kolibri-explore-plugin) manuq@endless-dev:template-ui$ yarn run lint
yarn run v1.22.10
$ vue-cli-service lint
 DONE  No lint errors found!
Done in 1.68s.
```

File: packages/template-ui/src/store/index.js (hunk 2 above)
Comment (manuq) [reply packages/template-ui/src/store/index.js:77]: I think we should do everything through APIs.

File: packages/template-ui/src/views/Search.vue [hunk 3]
```diff
@@ -79,8 +79,11 @@ export default {
   },
   methods: {
     search() {
-      this.resultNodes = this.searchNodes(this.cleanedQuery);
-      this.searching = false;
+      return window.kolibri.searchContent({ keyword: this.cleanedQuery, under: 'self' })
```
[the remark this answers is not in the document]
Comment (manuq) [reply packages/template-ui/src/views/Search.vue:82]: According to the doc that's what `self` does. But it doesn't matter, I also tried with a hardcoded ID like `"e11462f71c6f5472b113311c69071b05"` for Dance or even with "123" and it always fetches all nodes in all channels. I added this to the API test report.

**Comment by manuq** [comment]
I answered about the search comment above.

About the linter, I think we are fine, because it seems yours is a bit more restrictive, so your commit is also passing here and does not harm at all. Anyways it would be nice if we can unveil this mistery. Maybe you have a hidden yarn eslint file? 

Merging.

**Pull request merged**